include/seastar/core/semaphore.hh

```diff
@@ -86,6 +92,11 @@ public:
     }
 };
 
+#ifdef SEASTAR_DEADLOCK_DETECTION
+    template<typename ExceptionFactory = semaphore_default_exception_factory, typename Clock = typename timer<>::clock>
+    class semaphore_units;
+#endif
+
```

Comment:
I suggest moving this forward declaration near the declaration of `std::set`, to keep the changes more local.